### ISSUE TITLE
Short form for match filtering conditions

### ISSUE THREAD

**Issue by generall** [qdrant/qdrant#1196]
**Is your feature request related to a problem? Please describe.**
There are situations where multiple filtering values should be applied to the same field.
The filtering criteria should consist of multiple `should` clauses, which makes request inconvenient to work with. 

**Describe the solution you'd like**
Extend query syntax to allow checking multiple values in `match` condition.

**Describe alternatives you've considered**
Extend client DSL to support automatic conversion into `should` clauses

**Comment by coszio** [comment]
I can give this a shot if it hasn't been taken yet 😄

**Comment by coszio** [comment]
I think a good solution for this is to add the `any` keyword to enable syntax like this to be used within the `match` clause
```json
{
    "filter": {
        "should": [
            {
                "key": "city",
                "match": {
                    "any": [
                        "San Francisco",
                        "New York",
                        "Tokyo",
                    ]
                }
            }
        ]
    }
    ...
}

```

@generall what do you think? If it sounds great I can start working on this

**Comment by generall** [comment]
looks 👍

**Comment by HridayM25** [comment]
Is there any work yet to be done for this issue?

**Comment by coszio** [comment]
@HridayM25 I believe I only missed to implement the adapter for this on the python client

**Comment by HridayM25** [comment]
Can this please be assigned to me?
I am a beginner at open source contribution, and would love to contribute to this community.
I just need to follow what you have done right? (adding "any" to "match")

**Comment by HridayM25** [comment]
Can this please be assigned to me?
I am a beginner at open source contribution, and would love to contribute to this community.

**Comment by HridayM25** [comment]
Hello! I have tried something, please review it and let me know if I am going in the correct direction or not.

`class MatchAny(BaseModel):
    """
    Exact matches of the given list of strings or string
    """

    value: "ValueListVariants" = Field(..., description="Exact matches of the given list of values")
`

and added ValueListVariants as 

`ValueListVariants= Union[
    StrictBool,
    StrictInt, 
    StrictStr,
    List
]
`

I am not sure if this correct or not, I can change it according to the feedback.
Thanks

**Comment by generall** [comment]
> Hello! I have tried something, please review it and let me know if I am going in the correct direction or not.
> 
> ```
> class MatchAny(BaseModel):
>     """
>     Exact matches of the given list of strings or string
>     """
>     
>     value: "ValueListVariants" = Field(..., description="Exact matches of the given list of values")
> ```
> 
> 
> and added ValueListVariants as 
> 
> `ValueListVariants= Union[
>     StrictBool,
>     StrictInt, 
>     StrictStr,
>     List
> ]
> `
> 
> I am not sure if this correct or not, I can change it according to the feedback.
> Thanks

There is a script to update python structures from openAPI/gRPC, no need to change anything manually

**Comment by HridayM25** [comment]
Hi! 
Could you please give some guidance on how this could be done?
I have gone through gRPC and OpenAPI but am not getting on how to approach this further.

**Comment by coszio** [comment]
@HridayM25 you can look into the [tools folder](https://github.com/qdrant/qdrant_client/tree/master/tools) for these scripts

**Comment by coszio** [comment]
Included in v1.1.0
